frontend/src/pages/hardwareProfiles/nodeResource/NodeResourceForm.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { TextInput, FormGroup, Form } from '@patternfly/react-core';
+import { Identifier } from '~/types';
+import CountFormField from './CountFormField';
+import { validateDefaultCount, validateMinCount } from './utils';
+
+type NodeResourceFormProps = {
+  identifier: Identifier;
+  onUpdate: (data: Identifier) => void;
+};
+
+const NodeResourceForm: React.FC<NodeResourceFormProps> = ({ identifier, onUpdate }) => {
+  const handleFieldUpdate = (field: keyof Identifier, value: unknown) => {
+    onUpdate({ ...identifier, [field]: value });
+  };
+
+  return (
+    <Form>
+      <FormGroup label="Resource label" fieldId="resource-label">
+        <TextInput
+          value={identifier.displayName || ''}
+          onChange={(_, value) => handleFieldUpdate('displayName', value)}
+        />
+      </FormGroup>
+
+      <FormGroup label="Resource identifier" fieldId="resource-identifier">
+        <TextInput
+          value={identifier.identifier || ''}
+          onChange={(_, value) => handleFieldUpdate('identifier', value)}
+        />
+      </FormGroup>
+
+      <CountFormField
+        size={String(identifier.defaultCount) || ''}
+        setSize={(value: string) => handleFieldUpdate('defaultCount', value)}
+        label="Default"
+        fieldId="default"
+        errorMessage="Default must be equal to or between the minimum and maximum allowed limits."
+        isValid={validateDefaultCount(identifier)}
+      />
+
+      <CountFormField
+        size={String(identifier.minCount) || ''}
+        setSize={(value: string) => handleFieldUpdate('minCount', value)}
+        label="Minimum allowed"
+        fieldId="minimum-allowed"
+        isValid={validateMinCount(identifier)}
+        errorMessage="Minimum allowed value cannot exceed the maximum allowed value"
+      />
+
+      <CountFormField
+        size={String(identifier.maxCount) || ''}
+        setSize={(value: string) => handleFieldUpdate('maxCount', value)}
+        label="Maximum allowed"
+        fieldId="maximum-allowed"
+      />
```

Comment:
This count form field you created assumes everything uses the memory units. instead

CPU should use cpu units
memory use memory units
and other has no units

also there are already components for `MemoryField` and `CPUField`

Reply:
so you probably dont need a CountFormField